test/Common/TestHelpers.cs

```diff
@@ -196,9 +202,47 @@ public static DurableTaskOptions GetDurableTaskOptionsForStorageProvider(string
             }
         }
 
+        // Create a valid task hub from the test name, and add a random suffix to avoid conflicts
         public static string GetTaskHubNameFromTestName(string testName, bool enableExtendedSessions)
         {
-            return testName.Replace("_", "") + (enableExtendedSessions ? "EX" : "") + PlatformSpecificHelpers.VersionSuffix;
+            string strippedTestName = testName.Replace("_", "");
+            string truncatedTestName = strippedTestName.Substring(0, Math.Min(35, strippedTestName.Length));
+            string testPropertiesSuffix = (enableExtendedSessions ? "EX" : "") + PlatformSpecificHelpers.VersionSuffix;
+            string randomSuffix = Guid.NewGuid().ToString().Substring(0, 4);
+            return truncatedTestName + testPropertiesSuffix + randomSuffix;
+        }
+
+        internal static async Task DeleteAllElementsInStorageTaskHubAsync(string connectionString, string taskHub, int partitionCount)
+        {
+            var deletionTasks = new List<Task>();
+            CloudStorageAccount storageAccount = CloudStorageAccount.Parse(connectionString);
+
+            // Delete blobs
+            CloudBlobClient blobClient = storageAccount.CreateCloudBlobClient();
+            CloudBlobContainer leases = blobClient.GetContainerReference($"{taskHub}-leases");
+            deletionTasks.Add(leases.DeleteIfExistsAsync());
+            CloudBlobContainer largeContainers = blobClient.GetContainerReference($"{taskHub}-largemessages");
+            deletionTasks.Add(largeContainers.DeleteIfExistsAsync());
```

Comment:
We should also delete the {taskHub}-applease blob container.

But also, when is this method called? Can we just call a method in AzureStorageOrchestrationService instead? We would have to update this method whenever we add another container etc. Also, it's specific to AzureStorage as the backend.

Reply:
It turns out this actually isn't in use, it is left over from my approach for Durable V1 branch. 

I'll delete this method. The actual delete logic relies on the DeleteTaskHub method on the AzureStorageOrchestrationService.

In general, I am fine with it working only for Azure Storage, as we are discussing that each storage provider supported by Durable Functions will have some way to run these tests themselves, and they will need to figure out their own cleanup strategy.